bin-ranges-transfer/src/main/java/uk/gov/pay/java_lambdas/bin_ranges_transfer/App.java

```diff
@@ -0,0 +1,128 @@
+import uk.gov.pay.java_lambdas.bin_ranges_transfer.model.SftpDownload;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.config.Constants.AWS_ACCOUNT_NAME;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.config.Constants.PASSPHRASE_PARAMETER_NAME;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.config.Constants.PRIVATE_KEY_PARAMETER_NAME;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.config.Constants.S3_BUCKET_NAME;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.config.Constants.SFTP_DIRECTORY;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.config.Constants.SFTP_FILE_PREFIX;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.config.Constants.getSftpHost;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.config.Constants.getSftpPort;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.config.Constants.getSftpUsername;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.util.Utils.createTempPrivateKeyFile;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.util.Utils.extractDateString;
+import static uk.gov.pay.java_lambdas.bin_ranges_transfer.util.Utils.parameterRequestWithDecryption;
+
+public class App implements RequestHandler<Void, Object> {
+    private static final Logger logger = LoggerFactory.getLogger(App.class);
+    private final S3Client s3Client;
+    private final SshClient sshClient;
+    private final SsmClient ssmClient;
+
+    public App() throws GeneralSecurityException, IOException {
+        s3Client = DependencyFactory.s3Client();
+        ssmClient = DependencyFactory.ssmClient();
+        HashMap<String, String> params = getSsmParameters();
+        sshClient = DependencyFactory.sshClient(
+            createTempPrivateKeyFile(params.get("privateKey")),
+            params.get("passphrase")
+        );
+    }
+
+    @Override
+    public String handleRequest(Void input, final Context context) {
+        logger.info("fn: {}, version: {}.", context.getFunctionName(), context.getFunctionVersion());
+        sshClient.start();
+        try {
+            ClientSession session = sshClient.connect(getSftpUsername(), getSftpHost(), getSftpPort())
+                .verify()
+                .getSession();
+
+            session.auth().verify(10, TimeUnit.SECONDS);
+
+            try (SftpClient sftpClient = DependencyFactory.sftpClient(session)) {
+
+                SftpClient.Handle handle = sftpClient.openDir(SFTP_DIRECTORY);
+                List<SftpClient.DirEntry> dirEntries = sftpClient.readDir(handle);
+
+                List<SftpDownload> downloadList = dirEntries.stream()
+                    .filter(entry -> entry.getFilename().contains(SFTP_FILE_PREFIX))
```

Comment:
As we're looking for a prefix I'd suggest using `.startsWith`.

Reply:
Good spot, agreed this is confusing. Will update